x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/rules_table_context.tsx

```diff
@@ -375,6 +382,67 @@ export const useRulesTableContext = (): RulesTableContextType => {
 export const useRulesTableContextOptional = (): RulesTableContextType | null =>
   useContext(RulesTableContext);
 
+interface ExtendedData {
+  rules: ExtendedRule[];
+  total: number;
+}
+
+function useFindExtendedRules(...params: Parameters<typeof useFindRules>) {
+  const {
+    data: { rules, total } = { rules: [], total: 0 },
+    refetch: refetchRules,
+    ...restResult
+  } = useFindRules(...params);
+
+  // Fetch rule snooze settings
+  const {
+    data: rulesSnoozeSettings,
+    isError: isSnoozeSettingsFetchingError,
+    refetch: refetchSnoozeSettings,
+  } = useFetchRulesSnoozeSettings(
+    rules.map((x) => x.id),
+    { enabled: rules.length > 0 }
+  );
+
+  const extendedData = useMemo<ExtendedData>(() => {
+    if (isSnoozeSettingsFetchingError) {
+      return {
+        rules: rules.map((r) => ({
+          ...r,
+          snoozeSettings: new Error(i18n.UNABLE_TO_FETCH_RULE_SNOOZE_SETTINGS),
```

Comment:
Where do we handle this error?

Reply:
`useRuleSnoozeColumn` in `use_columns.tsx` checks for this error and displays a disabled bell icon with a tooltip if it's an error.